rspace/src/main/scala/coop/rchain/rspace/ReplayRSpace.scala

```diff
@@ -98,7 +98,11 @@ class ReplayRSpace[F[_]: Sync, C, P, A, K](
             )
     } yield r
 
-// TODO: refactor this monster
+  private def filterMatching(comm: COMM)(data: Seq[internal.Datum[A]]) =
+    data.zipWithIndex.filter {
```

Comment:
While we're disparaging `zipWithIndex`, should this not also be `seq.iterator.zipWithIndex...toSeq`?

Reply:
OMG, it slipped through the cracks. Probably when I was merging my branches. Thanks for noticing!

Reply:
Fxied in fb0526681397ec4e067344109bb88797ea6c568d